node/src/components/storage.rs

```diff
@@ -713,6 +713,12 @@ impl Storage {
     ) -> Result<Option<BlockSignatures>, Error> {
         Ok(tx.get_value(self.block_metadata_db, block_hash)?)
     }
+
+    /// Get the lmdb environment
+    #[cfg(test)]
+    pub(crate) fn env(&self) -> &Environment {
```

Comment:
This perhaps means we should set the value of `MAX_TRANSACTIONS` greater than `1` for `#[cfg(test)]`, since we could now be creating new transactions in the test while the node under test has also created one?

I'm not sure though.  I don't think that's possible with the tests you've added in this PR, and maybe if this happens in future tests, the error will be obvious enough to fix.